platform/mbed_wait_api_rtos.cpp

```diff
@@ -31,23 +31,39 @@ void wait(float s)
 
 void wait_ms(int ms)
 {
-    wait_us(ms * 1000);
+    // Wait equivalent to 1 ms near the tick might result into zero wait.
+    // Hence ms precision can be used for > 1 ms wait only.
+    if ((ms >= 1) && core_util_are_interrupts_enabled()) {
+        ThisThread::sleep_for(ms);
+    } else {
+        wait_us(ms * 1000);
+    }
 }
 
 void wait_us(int us)
 {
-    const ticker_data_t *const ticker = get_us_ticker_data();
-
-    uint32_t start = ticker_read(ticker);
-    if ((us >= 1000) && core_util_are_interrupts_enabled()) {
+    uint32_t ms = 0;
+    uint32_t sub_ms = us;
+    // Wait > 10ms will have ms precision, else us
+    if ((us >= 10000) && core_util_are_interrupts_enabled()) {
         // Use the RTOS to wait for millisecond delays if possible
-        sleep_manager_lock_deep_sleep();
-        Thread::wait((uint32_t)us / 1000);
-        sleep_manager_unlock_deep_sleep();
+        ms = (us / 1000);
+        sub_ms -= (ms * 1000);
+        // Round up the ms wait to avoid busy wait loop
+        if (sub_ms > 500) {
+            ms++;
+            sub_ms = 0;
+        }
+        ThisThread::sleep_for(ms);
+    }
+
+    if (sub_ms > 0) {
```

Comment:
Could this be the `else` case of the above code? That way delays above the threshold never busy wait.

Reply:
They still don;t in current logic since we set `sub_ms = 0;` 
But i see your reduced instruction code with above comment `ms = ((us + 500) / 1000);` will update as per that